src/plugins/intel_gna/src/avx2.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "avx2.hpp"
+
+#if defined(HAVE_AVX2)
+
+#    include <immintrin.h>
+
+#    include "preprocessing.hpp"
+
+void GNAPluginNS::ConvertMatrixFp32ToInt16(int16_t* ptr_dst,
```

Comment:
The implementations loos very similar, maybe it would be possible to create one template method?